### PR TITLE
[ CI TODOs blocking merge ] Add E2E test for vSphere management + workload cluster

### PR DESCRIPTION
**Pull request by karuppiah7890**
## What this PR does / why we need it

Overview of what's been done in the PR
- Introduce vSphere management and workload cluster E2E test script
- Check the time of management cluster creation, workload cluster creation, management cluster deletion and workload cluster deletion
- Add docs for vSphere management and workload cluster E2E test
- Some small improvements to vSphere standalone cluster E2E test script
- Check the time of standalone cluster creation, standalone cluster deletion

## Details for the Release Notes

```release-note
Add E2E test for vSphere management + workload cluster
```

## Which issue(s) this PR fixes

There is no issues to track vSphere E2E test for management + workload cluster
This does fix one other small generic issue #1406 

## Describe testing done for PR

Standalone cluster with `e850577cffffed6a81bf0ddc7eb835c3dbaa826d` - https://github.com/karuppiah7890/community-edition/runs/3536047637?check_suite_focus=true

Management cluster + Workload cluster with `e850577cffffed6a81bf0ddc7eb835c3dbaa826d` - https://github.com/karuppiah7890/community-edition/runs/3536047636?check_suite_focus=true

Some recent tests done with `v0.8.0-rc.2` and `v0.8.0-rc.3` TCE builds -

vSphere Management + Workload Cluster E2E test using TCE `v0.8.0-rc.2` - https://github.com/karuppiah7890/community-edition/runs/3631760197?check_suite_focus=true

vSphere Standalone Cluster E2E test using TCE `v0.8.0-rc.2` - https://github.com/karuppiah7890/community-edition/runs/3631964927?check_suite_focus=true

vSphere Management + Workload Cluster E2E test using TCE `v0.8.0-rc.3` - https://github.com/karuppiah7890/community-edition/runs/3632139015?check_suite_focus=true 

vSphere Standalone Cluster E2E test using TCE `v0.8.0-rc.3` -
https://github.com/karuppiah7890/community-edition/runs/3632263180?check_suite_focus=true 
## Special notes for your reviewer

Most of the notes are present in the markdown doc file which is part of this PR and is meant to serve as a doc for anyone trying to understand the E2E test without reading and understanding the code which can take lots of time

Before merging the PR -
- We need to add two secrets - `MANAGEMENT_CLUSTER_VSPHERE_CONTROL_PLANE_ENDPOINT`, `WORKLOAD_CLUSTER_VSPHERE_CONTROL_PLANE_ENDPOINT`
- We need to remove some secrets - `JUMPER_SSH_HOST_IP`, `JUMPER_SSH_USERNAME`, `JUMPER_SSH_PRIVATE_KEY`, `JUMPER_SSH_KNOWN_HOSTS_ENTRY`
- We need to create a self hosted runner and register it in this repo - I can create the runner, but I'll need help with registering to the repo as I need a runner token from the repo settings, and we need to label the runner with the label `vsphere-e2e-runner`. We use `vsphere-e2e-runner` label in the workflow config yaml too

### CODE REVIEW

**Review by rajaskakodkar — CHANGES_REQUESTED** [review]

**Review by davidvonthenen — COMMENTED** [review]
I think I'm following the logic here but am looking forward to the details of the runner setup. It seems like if you can guarantee that the runner environment will not "corrupt" or impact other future runs so that jobs can run concurrently on the single runner, this should be good. When you fill out the details in your comment here https://github.com/vmware-tanzu/community-edition/pull/1554/#issuecomment-914522598

**Review by davidvonthenen — COMMENTED** [review]
Other than those 2 changes this looks good! After making the changes, squash, and fix the commit message to a single message. Awesome!

**Review by rajaskakodkar — APPROVED** [review]

File: .github/workflows/e2e-vsphere-standalone-cluster.yaml [hunk 1]
```diff
@@ -5,28 +5,20 @@ on:
...
-          sudo rm -rf "/usr/local/share/boost"
-          df -h
-
       - name: Check out code
         uses: actions/checkout@v1
 
-      - name: Run TCE vSphere Standalone Cluster E2E Test
+      - name: Run vSphere Standalone Cluster E2E Test
         env:
           GITHUB_TOKEN: ${{ secrets.GH_E2E_ACCESS_TOKEN }}
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply .github/workflows/e2e-vsphere-standalone-cluster.yaml:23]: Since we are public now... can we switch these tokens?

**Review by davidvonthenen — CHANGES_REQUESTED** [review]
Still need to configure the runner `vsphere-e2e-runner`. Did you need help with that? It would be better if you could handle this because I haven't logged into the infrastructure to add the node in the list of runners.

**Comment by karuppiah7890** [comment]
https://github.com/vmware-tanzu/community-edition/pull/1554#issuecomment-930045764 , yeah, I'll need help with getting the token from the repo since I don't have access to the settings.

**Comment by karuppiah7890** [comment]
Looks like I need to recreate this PR since the re-fork after TCE repo went public. Gonna close this one

**Pull request closed without merging**